### DISCUSSION TITLE
Run Commands

### DISCUSSION THREAD

**Discussion by sigzero** [orbitalquark/textadept#25] [General]
I have the following for bash/shell scripts in my init.lua:

    textadept.run.run_commands.bash = 'C:/Bin/shellcheck "%f"'
    textadept.run.compile_commands.bash = 'C:/Bin/shellcheck "%f"'

I added the second just as a test. The first does not work. TA continues to use "sh" for the run_command. The second one does work as expected.

According to the manual (https://orbitalquark.github.io/textadept/manual.html#compile-run-and-build) the first line should work.

**Comment by rgieseke** [comment]
What is your file named?

Extension comes before the lexer:

textadept.run.run(filename)

Runs file filename or the current file using an appropriate shell command from the run_commands table. The shell command is determined from the file’s filename, extension, or language in that order.

**Comment by sigzero** [comment]
The file is named "shcsv.sh".

**Comment by rgieseke** [comment]
Then i would guess, the file extension rule has precedence. Haven't checked the definitions though.

**Comment by rgieseke** [comment]
Seems shellcheck works for different shells so there's probablt no problem using `.sh` for the definition.

**Reply by sigzero** [reply]
Right, the second "compile" command works as expected with shellcheck.

**Comment by orbitalquark** [comment]
Robert is correct. The 'sh' extension has precedence over the 'bash' lexer, so you would have to use

    textadept.run.run_commands.sh = '...'

The compile command works because there is no `textadept.run.compile_commands.sh` defined, so `textadept.run.compile_commands.bash` is tried afterwards, and found.
